my-app/src/app/courses-page/course-search/course-search.component.ts

```diff
@@ -1,18 +1,39 @@
-import { ChangeDetectionStrategy, Component, EventEmitter, Output } from '@angular/core';
+import { ChangeDetectionStrategy, Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { of, Subject, Subscription } from 'rxjs';
+import { debounceTime, delay, distinctUntilChanged, filter, map, mergeMap } from 'rxjs/operators';
 
 @Component({
   selector: 'app-course-search',
   templateUrl: './course-search.component.html',
   styleUrls: ['./course-search.component.css'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class CourseSearchComponent {
-  inputValue: string;
+export class CourseSearchComponent implements OnInit, OnDestroy {
+  // inputValue: string;
+
+  private subscriptions: Subscription = new Subscription();
+
+  public keyUp = new Subject<KeyboardEvent>();
 
   @Output() searchPhrase = new EventEmitter<string>();
 
-  onClick(): void {
-    this.searchPhrase.emit(this.inputValue);
-  }
+  public ngOnInit(): void {
+    this.subscriptions.add(
+      this.keyUp.pipe(
+        map((event: any) => event.target.value),
+        filter((res) => res.length > 2),
```

Comment:
what if I want to clear the search? This stream won't emit the value and the search won't be triggered.